src/background/services/keyAutoAdd.ts

```diff
@@ -178,25 +178,49 @@ export class KeyAutoAddService {
       return false;
     }
   }
+
+  static async registerContentScripts({ browser }: Pick<Cradle, 'browser'>) {
+    const scripting = browser.scripting;
+    const existingScripts = await scripting.getRegisteredContentScripts();
+    const existingScriptIds = new Set(existingScripts.map((s) => s.id));
+    const scripts = getContentScripts().filter(
+      (s) => !existingScriptIds.has(s.id),
+    );
+    await scripting.registerContentScripts(scripts);
+  }
+}
+
+function getContentScripts(): Scripting.RegisteredContentScript[] {
+  return [
+    {
+      id: 'keyAutoAdd/testWallet',
+      matches: [
+        'https://wallet.interledger-test.dev/*',
+        'https://wallet.interledger.cards/*',
+      ],
+      js: ['content/keyAutoAdd/testWallet.js'],
+      runAt: 'document_end',
```

Comment:
Any particular reason we are doing this at `document_end` ?

Reply:
Oh, `document_idle` is a better choice. I though end was fired later 🤦🏽